pkg/controller/clusterdeprovision/awsactuator.go

```diff
@@ -0,0 +1,62 @@
+package clusterdeprovision
+
+import (
+	log "github.com/sirupsen/logrus"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/ec2"
+
+	corev1 "k8s.io/api/core/v1"
+
+	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1"
+	awsclient "github.com/openshift/hive/pkg/awsclient"
+)
+
+func init() {
+	registerActuator(&awsActuator{awsClientFn: getAWSClient})
+}
+
+// Ensure AWSActuator implements the Actuator interface. This will fail at compile time when false.
+var _ Actuator = &awsActuator{}
+
+// AWSActuator manages getting the desired state, getting the current state and reconciling the two.
+type awsActuator struct {
+	// awsClientFn is the function to build an AWS client, here for testing
+	awsClientFn func(*hivev1.ClusterDeprovision, client.Client, log.FieldLogger) (awsclient.Client, error)
+}
+
+type awsClientBuilderType func(secret *corev1.Secret, region string) (awsclient.Client, error)
+
+// CanHandle returns true if the actuator can handle a particular ClusterDeprovision
+func (a *awsActuator) CanHandle(clusterDeprovision *hivev1.ClusterDeprovision) bool {
+	return clusterDeprovision.Spec.Platform.AWS != nil
+}
+
+// TestCredentials ensures that the the aws credentials are usable.
+func (a *awsActuator) TestCredentials(clusterDeprovision *hivev1.ClusterDeprovision, c client.Client, logger log.FieldLogger) error {
+	awsClient, err := a.awsClientFn(clusterDeprovision, c, logger)
+	if err != nil {
+		return err
+	}
+
+	_, err = awsClient.DescribeInstances(&ec2.DescribeInstancesInput{DryRun: aws.Bool(true)})
```

Comment:
Perhaps I'm missing some context here. But why is the ability to DescribeInstances() enough for a deprovision? Are we only catching creds that have been revoked completely?
Have you considered an iam.GetUser() or sts.GetCallerIdentity() call if all you want to know is whether this ACCESS_KEY/SECRET_ACCESS_KEY pair are functional?